### DISCUSSION TITLE
Why go-lang? Isn't go really slow?

### DISCUSSION THREAD

**Discussion by kendofriendo** [anyproto/contributors#130] [Q&A] [answered]
Just curious

**Comment by requilence** [comment, marked as answer]
Hey there!
We chose Go back in 2019 year for a few reasons, even though some folks might think it's a bit slower compared to languages like C++ or Rust.

- Easy to Understand: Go is super straightforward and easy to read. This makes our code more maintainable and easier for new contributors to understand and get involved. 
- Great for Concurrency: Go has awesome features for handling lots of things at once (like managing multiple network connections), which is super important for our app.
- Cross-Platform Compatibility: Go works really well across different platforms. This means we can use the same codebase for our desktop and mobile apps, which saves us a lot of time and effort.  
- Strong Standard Library: Go comes with a lot of built-in functionality, especially for networking. This helps us a lot since our app relies heavily on network communication.
- Performance: While Go might not be the fastest language in every scenario, it's a compiled language and heavily optimized and fast enough for our needs. It strikes a good balance between speed and ease of use. For the most heavy-loaded code we pay more attention on working with memory and optimise code with benchmarks.
- Community and Support: Go has a great community and lots of resources and packages. It also actively developing improving performance and introducing requested features.

So, while Go might not be the absolute fastest, it's fast enough and brings a lot of other benefits to the table that make it a great fit for our project. Hope this helps clear things up! 😊
